build/Microsoft.DotNet.Cli.Compile.targets

```diff
@@ -29,7 +29,7 @@
 
     <ItemGroup>
       <CompileStageInputs Include="$(SrcDirectory)/**/*.cs" Exclude="$(SrcDirectory)/**/obj/**/*.cs" />
-      <CompileStageInputs Include="$(SrcDirectory)/**/project.json;$(SrcDirectory)/**/project.lock.json" />
+      <CompileStageInputs Include="$(SrcDirectory)/**/*.csproj;$(SrcDirectory)/**/project.assets.json" />
```

Comment:
That is what it does. Given how extensible the build is for each project in the solution, it can't do anything better. Up to you if you can and it's worth maintaining.